packages/web/src/components/track-flair/TrackFlair.tsx

```diff
@@ -0,0 +1,62 @@
+import { ReactNode, RefObject } from 'react'
+
+import { useTrack } from '@audius/common/api'
+import { ID } from '@audius/common/models'
+import cn from 'classnames'
+
+import CoSignFlair from './CosignFlair'
+import styles from './Flair.module.css'
+import { Size } from './types'
+
+interface TrackFlairProps {
+  size: Size
+  children: ReactNode
+  className?: string
+  hideToolTip?: boolean
+  id: ID
+  forwardRef?: RefObject<HTMLDivElement>
+}
+
+const TrackFlair = (props: TrackFlairProps) => {
+  const { forwardRef, size, children, className, id, hideToolTip } = props
+  const { data: track } = useTrack(id)
+
+  if (!track) return
+
+  const remixTrack = track.remix_of?.tracks[0]
+  const hasRemixAuthorReposted = remixTrack?.has_remix_author_reposted ?? false
+  const hasRemixAuthorSaved = remixTrack?.has_remix_author_saved ?? false
+
+  const isCosign = hasRemixAuthorReposted || hasRemixAuthorSaved
+
+  const flair = isCosign ? (
+    <CoSignFlair
+      coSignName={remixTrack?.user.name}
```

Comment:
@DejayJD a bit weird we have user deeply nested here, just an fyi on the normalization front.